pacman/operations/partition_algorithms/partition_and_place_partitioner.py

```diff
@@ -100,9 +105,14 @@ def __call__(
 
             # if not, partition
             if machine_vertices is None:
-                self._partition_vertex(
-                    vertex, plan_n_timesteps, machine_graph, graph_mapper,
-                    resource_tracker, progress, vertex_groups)
+                if isinstance(vertex, SplitterByAtoms):
```

Comment:
What about application vertices that are not `SplitterByAtoms` or `HandOverToVertex`?

Reply:
if your saying we need an extra else to handle the case. then im ok with that. complaining about the 2 apis. there ARE two apis at this point. ones which partition themsleves, and ones which follow the old rule of letting the partitioner partition them via sdram dtcm and cpu cycles logics on slices

Reply:
there are no applications which are not of those 2 types. 

> Why the fuck is this code making it possible for application vertices to _not_ be responsible for creating their associated machine vertex?

coz there are ones. like per say. anything going throguh AbstractPopulationVertex

Reply:
My suggestion would be to make *all* application vertices partition themselves.  I suspect the current partitioner is overkill for what actually happens in most cases, which is that the neurons of a neural simulation are divided into groups defined by the neurons-per-core values.  So for most vertices they can just do this splitting simply and everything will likely work.